opentracing-shim/src/opentracingshim/__init__.py

```diff
@@ -0,0 +1,241 @@
+        self._otel_span.add_lazy_event(event)
+
+        # Return self for call chaining.
+        return self
+
+    def set_baggage_item(self, key, value):
+        raise NotImplementedError
+        # TODO: Implement.
+
+    def get_baggage_item(self, key):
+        raise NotImplementedError
+        # TODO: Implement.
+
+    # TODO: Verify calls to deprecated methods `log_event()` and `log()` on
+    # base class work properly (it's probably fine because both methods call
+    # `log_kv()`).
+
+
+class ScopeWrapper(opentracing.Scope):
+    def close(self):
+        self._span.finish()
+        # TODO: Set active span on OpenTelemetry tracer.
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
+
+
+class ScopeManagerWrapper(opentracing.ScopeManager):
+    def __init__(self, tracer):
+        # The only thing the `__init__()` method on the base class does is
+        # initialize `self._noop_span` and `self._noop_scope` with no-op
+        # objects. Therefore, it doesn't seem useful to call it.
+        # pylint: disable=super-init-not-called
+        self._tracer = tracer
+
+    @contextmanager
+    def activate(self, span, finish_on_close):
+        with self._tracer.unwrap().use_span(
+            span.unwrap(), end_on_exit=finish_on_close
+        ) as otel_span:
+            wrapped_span = SpanWrapper(
+                self._tracer, otel_span.get_context(), otel_span
+            )
+            yield ScopeWrapper(self, wrapped_span)
+
+    @property
+    def active(self):
+        span = self._tracer.get_current_span()
+        wrapped_span = SpanWrapper(self._tracer, span.get_context(), span)
+        return ScopeWrapper(self, wrapped_span)
+        # TODO: Return a saved instance of SpanWrapper instead of constructing
+        # a new object (and the same for ScopeWrapper?).
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
+
+
+class TracerWrapper(opentracing.Tracer):
+    def __init__(self, tracer, scope_manager=None):
+        self._otel_tracer = tracer
+
+        # If a scope manager isn't provided by the user, create a
+        # `ScopeManagerWrapper` instance and use it to initialize the
+        # `TracerWrapper`.
+        if scope_manager is None:
+            scope_manager = ScopeManagerWrapper(self)
+
+        super().__init__(scope_manager=scope_manager)
+
+    def unwrap(self):
+        """Returns the wrapped OpenTelemetry `Tracer` object."""
+
+        return self._otel_tracer
+
+    @property
+    def active_span(self):
+        span = self._otel_tracer.get_current_span()
+        if span is None:
+            return None
+        return SpanWrapper(self, span.get_context(), span)
+        # TODO: Return a saved instance of SpanWrapper instead of constructing
+        # a new object.
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
+
+    @contextmanager
```

Comment:
Please don't use `@contextmanager` but return the result of `activate` directly without using `with` or `yield`. This is more efficient and simpler *and* required to be correct if you read the docs of `start_active_span`: https://github.com/opentracing/opentracing-python/blob/7d2e62b5999d3cdec9a9c2720a34a1d11f425461/opentracing/tracer.py#L70:

> Returns a newly started and activated :class:`Scope`.

Currently, this doesn't return a Scope, but some anonymous ContextManager object.

Reply:
This is fixed now.